### PR TITLE
Stunt

### PR DESCRIPTION
**Pull request by Patharapol1573**

### CODE REVIEW

File: src/states/SelectMoveState.py [hunk 1]
```diff
@@ -67,6 +68,20 @@ def Enter(self, params):
...
+            for index in range(len(self.avilableMoveTile)):
+                if not self.field[self.avilableMoveTile[index]].is_occupied():
+                    if self.avilableMoveTile[index] <= self.enemy.fieldTile_index:
+                        randomLeft.append(index)
+                    if self.avilableMoveTile[index] >= self.enemy.fieldTile_index:
+                        randomRight.append(index)
+            if self.enemy.fieldTile_index > self.player.fieldTile_index:
+                self.selectMoveTile = random.choice(randomLeft)
+            else:
+                self.selectMoveTile = random.choice(randomRight)
```
Comment (phukaokub) [inline src/states/SelectMoveState.py:84]: What a pure random move hahaha. Nice one.

**Review by phukaokub — APPROVED** [review]

File: src/states/BattleInitialState.py [hunk 2]
```diff
@@ -22,6 +22,9 @@ def Enter(self, params):
         self.turn = params['turn']
         self.currentTurnOwner = params['currentTurnOwner']  
 
+        self.player.stunt = False
+        self.enemy.stunt = False
```
Comment (krittintrs) [inline src/states/BattleInitialState.py:26]: shouldn't this be set inside the construction method of Entity class?

File: src/states/BattleResolveState.py [hunk 3]
```diff
@@ -90,134 +90,141 @@ def update(self, dt, events):
...
+                case EffectType.SPAWN:
+                    pass
+                case EffectType.HEAL:
+                    pass
+                case EffectType.COPY:
+                    pass
+                case _:
+                    print(f'Unknown effect type: {effect.type}')
+        else:
+            print('stunt')
```
Comment (krittintrs) [inline src/states/BattleResolveState.py:219]: is there any visual feedback like rendering stunt message on the game? or is it just printing on the terminal?

File: src/states/SelectAttackState.py [hunk 4]
```diff
@@ -103,34 +109,37 @@ def update(self, dt, events):
...
-                                        self.enemy.add_buffs(buffList)
-                                else:
-                                    # ATTACK BLOCK
-                                    gSounds['block'].play()
-                                    print(f'{self.field[self.avilableAttackTile[self.selectAttackTile]].entity} takes no damage')
-                                self.field[self.avilableAttackTile[self.selectAttackTile]].entity.print_stats()
                             else:
-                                print("no entity on the targeted tile")
+                                self.enemy.ChangeAnimation("attack")
+                                self.player.ChangeAnimation("knockdown")
```
Comment (krittintrs) [inline src/states/SelectAttackState.py:120]: should the animation be played only when the attack is hit? I think there are two ways to play the animation
- the attack animation is played immediately / the knockdown is played when attack is hit
- both attack and knockdown animation will be played only when attack is hit

File: src/states/SelectAttackState.py [hunk 5]
```diff
@@ -103,34 +109,37 @@ def update(self, dt, events):
...
+                            damage = self.player.attack - self.field[self.avilableAttackTile[self.selectAttackTile]].entity.defense
```
Comment (krittintrs) [inline src/states/SelectAttackState.py:121]: the damage should calculate differently between different effectOwner

File: src/states/SelectAttackState.py [hunk 6]
```diff
@@ -103,34 +109,37 @@ def update(self, dt, events):
...
+                                gSounds['attack'].play()
+                                self.field[self.avilableAttackTile[self.selectAttackTile]].entity.health -= damage
+                                self.field[self.avilableAttackTile[self.selectAttackTile]].entity.stunt = True
+                                print(f'{self.field[self.avilableAttackTile[self.selectAttackTile]].entity} takes {damage} damage')
+                                if self.effect.type == EffectType.ATTACK_SELF_BUFF:
+                                    buffList = self.getBuffListFromEffect(self.effect)
+                                    self.player.add_buffs(buffList)
+                                if self.effect.type == EffectType.ATTACK_OPPO_BUFF:
+                                    buffList = self.getBuffListFromEffect(self.effect)
+                                    self.enemy.add_buffs(buffList)
```
Comment (krittintrs) [inline src/states/SelectAttackState.py:133]: this line also different between different effectOwner

File: src/states/SelectMoveState.py (hunk 1 above)
Comment (krittintrs) [reply src/states/SelectMoveState.py:84]: this could be use for confusion debuff in the future

File: src/states/SelectAttackState.py (hunk 4 above)
Comment (Patharapol1573) [reply src/states/SelectAttackState.py]: the attack animation is played immediately / the knockdown is played when attack is hit

File: src/states/BattleResolveState.py (hunk 3 above)
Comment (Patharapol1573) [reply src/states/BattleResolveState.py:219]: at the moment, it is just printing on the terminal.

**Review by rymTooo — APPROVED** [review]

**Review by phukaokub — APPROVED** [review]

File: src/states/BattleResolveState.py (hunk 3 above)
Comment (krittintrs) [reply src/states/BattleResolveState.py:219]: ok, might leave it as future improvement kub

**Review by krittintrs — APPROVED** [review]

**Pull request merged**